plugins/org.locationtech.udig.core/src/org/locationtech/udig/core/internal/FeatureUtils.java

```diff
@@ -641,67 +641,80 @@ private static boolean toLinearRing( Geometry geom, Class< ? extends Geometry> t
-
-		assert featureType != null : "featureType cannot be null";
-		assert propertyName != null : "propertyName != cannot be null";
-
-		//if an exact match is found return it
-		for (int i = 0; i < featureType.getAttributeCount(); i++) {
-			String name = featureType.getDescriptor(i).getLocalName();
-			if (propertyName.equals(name)) {
-				return name;
-			}
-		}
-		
-		//otherwise return the first match found by performing a case 
-		//insensitive check for equality 
-		for (int i = 0; i < featureType.getAttributeCount(); i++) {
-			String name = featureType.getDescriptor(i).getLocalName();
-			if (propertyName.equalsIgnoreCase(name)) {
-				return name;
-			}
-		}
-		return null;
-	}
+    /**
+     * Performs a lookup for the provided propertyName in the {@link FeatureType}. Matching process
+     * finds case sensitive attribute first, after that case insensitive comparison returns the
+     * first matching attribute
+     *
+     * @param featureType FeatureType to find named attribute
+     * @param propertyName attribute name
+     * @return the actual propertyName (case sensitive) or null if none is found
+     * @throws IllegalArgumentException if featureType and/or property is null/empty
+     */
+    public static String getActualPropertyName(final SimpleFeatureType featureType,
+            final String propertyName) {
+
+        if (featureType == null) {
+            throw new IllegalArgumentException("featureType cannot be null");
+        }
+        if (propertyName == null) {
+            throw new IllegalArgumentException("propertyName cannot be null");
+        }
+
+        // if an exact match is found return it
+        for (int i = 0; i < featureType.getAttributeCount(); i++) {
+            String name = featureType.getDescriptor(i).getLocalName();
+            if (propertyName.equals(name)) {
+                return name;
+            }
+        }
+
+        // otherwise return the first match found by performing a case
+        // insensitive check for equality
+        for (int i = 0; i < featureType.getAttributeCount(); i++) {
+            String name = featureType.getDescriptor(i).getLocalName();
+            if (propertyName.equalsIgnoreCase(name)) {
+                return name;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Performs a lookup for the provided propertyNames in the {@link FeatureType} using
+     * {@link #getActualPropertyName(SimpleFeatureType, String)} approach
+     *
+     * @param featureType  FeatureType to find named attribute
+     * @param propertyName List of Properties to check
+     * @return a list of actual propertyName (case sensitive). If a propertyName is not present then
+     *         it is skipped from the list
+     * @throws IllegalArgumentException if featureType and/or propertyNames is null
+     */
+    public static List<String> getActualPropertyName(final SimpleFeatureType featureType,
```

Comment:
As explained in a previous comment it is not current used but I included there since FeatureUtils is a genric util class and it might be handly for somebody in the future. After all it is a more generialized version of the one I am using. Checking udig code, it seems there are also similar cases in other util classes within udig project